core/src/memory.rs

```diff
@@ -44,6 +45,28 @@ pub struct MemoryInterval {
     pub size: u32,
 }
 
+impl MemoryInterval {
+    /// Convert `MemoryInterval` to `[u8; 8]` bytes.
+    /// `0..4` - `offset`
+    /// `4..8` - `size`
+    pub fn to_bytes(&self) -> [u8; 8] {
+        let mut bytes = [0u8; 8];
+        LittleEndian::write_u32(&mut bytes[0..4], self.offset);
+        LittleEndian::write_u32(&mut bytes[4..8], self.size);
+        bytes
+    }
+
+    /// Convert `[u8; 8]` bytes to `MemoryInterval`.
+    /// `0..4` - `offset`
+    /// `4..8` - `size`
+    pub fn from_bytes(bytes: &[u8]) -> Self {
+        assert!(bytes.len() == 8);
+        let offset = LittleEndian::read_u32(&bytes[0..4]);
```

Comment:
We could avoid using `LittleEndian` here, I guess
```suggestion
        let offset = u32::from_le_bytes(bytes[0..4].try_into().unwrap());
```